### PR TITLE
Generalize communicate_locally_moved_vertices() 

### PR DESCRIPTION
**Pull request by peterrum**
...for arbitrary distributed meshes. The new implementation is now based on `GridTools::exchange_cell_data_to_ghosts`.

I have been working on this together with @ifumagalli. Could you try the new function out in lifex if it indeed works for you.

### CODE REVIEW

**Comment by ifumagalli** [comment]
Thank you very much, @peterrum !
I tested it both on p::distributed::T and p::fullydistributed::T and it works perfectly!

File: source/distributed/tria_base.cc [hunk 1]
```diff
@@ -527,6 +527,56 @@ namespace parallel
...
+          vertices[v] = cell->vertex(v);
+        else
+          vertices[v] = invalid_point;
+
+      return vertices;
+    };
+
+    const auto unpack = [&](const auto &cell, const auto &vertices) {
+      for (const auto v : cell->vertex_indices())
+        if (std::isnan(vertices[v][0]) == false)
```
Comment (masterleinad) [inline source/distributed/tria_base.cc:566]:
```suggestion
        if (vertices[v] != invalid_point)
```
for symmetry with `pack`.

**Review by masterleinad — COMMENTED** [review]
Looks good to me. Just a minor suggestion.

File: source/distributed/tria_base.cc (hunk 1 above)
Comment (peterrum) [reply source/distributed/tria_base.cc:566]: This is not working. I have tried it out.

**Comment by peterrum** [comment]
Has anyone an idea how to tackle the (is already defined) error?

File: source/distributed/tria_base.cc (hunk 1 above)
Comment (masterleinad) [reply source/distributed/tria_base.cc:566]: What do you mean by "not working"?

**Comment by masterleinad** [comment]
I pushed a fix.

File: source/distributed/tria_base.cc (hunk 1 above)
Comment (peterrum) [reply source/distributed/tria_base.cc:566]: I get a `Floating point exception`.

**Comment by peterrum** [comment]
@masterleinad Thanks for the fix! Ready from my side!

File: source/distributed/tria_base.cc (hunk 1 above)
Comment (masterleinad) [reply source/distributed/tria_base.cc:566]: Hmm... Do you have a particular test that fails?

**Review by masterleinad — APPROVED** [review]

**Comment by peterrum** [comment]
@masterleinad All `ctest -R mpi/communicate_moved_vertices` tests run parallel.

**Pull request merged**